src/Misc/fileTools.ts

```diff
@@ -493,8 +518,44 @@ export class FileTools {
     public static IsFileURL(): boolean {
         return typeof location !== "undefined" && location.protocol === "file:";
     }
+
+    /**
+     * Test if the given uri is a base64 string
+     * @param uri The uri to test
+     * @return True if the uri is a base64 string or false otherwise
+     */
+    public static IsBase64DataUrl(uri: string): boolean {
+        // Check that the length of the string is at least as long as the minimum base64 data url length
+        if (uri.length >= 13) // "data:;base64,".length
+        {
+            if (uri.substr(0, 5) === "data:") {

```

Comment:
Without creating an extra string:
```suggestion
            if (uir.indexOf('data:') === 0) {
```

Reply:
This is true, but it will search the entire string if `data:` is not present. Probably not a big deal since big strings probably start with `data:`

Reply:
so use startsWith? :-)
Or use a regexp. avoid creating the extra object

Reply:
Yup. We did that below. Using startsWith is a good idea.